config/routes.rb

```diff
@@ -20,13 +20,12 @@ def matches?(request)
     end
 
     constraints account: /[^\/\?]+/ do
-      get  '/authn/:account/login' => 'credentials#login'
```

Comment:
Does this mean we're going to have to update our Conjur test files across various project to define the `authn` authenticator in the `CONJUR_AUTHENTICATORS` env var if we want to continue using the CLI to authenticate when another authenticator is present?

For example, in the k8s deploy scripts, we define `authn-k8s` but currently that doesn't prevent authentication using the Conjur CLI from working. If I'm understanding this correctly, we would now need to define `authn-k8s` AND `authn` to allow authentication both through the k8s authenticator and CLI?

Reply:
Also, what was the impetus for this change? Is there a reason we've decided to move things around?

Reply:
The immediate cause for these changes is to support allowing users to login to the UI using LDAP credentials. We explored a few different options for this, but most of them had one design smell or another. In the end we settled on a principal that resonates well with the existing design: Humans login, services authenticate.

So what this intends to lay out is a way in which authenticators may support either use case, where the overall flow (`User` -> `Login` -> `Authenticate`, or `Host` -> `Authenticate`) doesn't change, and token/key between login and authenticate is consistent enough that the UI and other end-user tools can support it without a special case for each authenticator.

You are correct that this PR would currently require you to include `authn` in the whitelisted authenticators if you also use another. I would be on board with including it behind the scenes with the enabled login authenticators to make this more fully backwards compatible.

Reply:
Okay, I think it's probably better to require users to explicitly include `authn` to keep the functionality because otherwise we would have to treat disabling `authn` as a special case, which sounds annoying.

The only trouble is that i's going to become important to communicate in docs that you need to include `authn` in `CONJUR_AUTHENTICATORS` in order to continue using the CLI w/ Conjur. Otherwise people might get a bit confused when they try to enable the k8s authenticator and can no longer use the CLI to load policy or something.

Any thoughts on this @jonahx ?

Reply:
To summarize our discussion in slack here: https://conjurhq.slack.com/archives/CA8AT283D/p1537367685000100

This would already be an issue when logging in with the CLI when a non-default authenticator is configured. For the built-in Conjur login flow to work, `authn` would need to be present in the enabled authenticators (`CONJUR_AUTHENTICATORS` env variable).

Reply:
Per slack convo, this has been the case for a while and was the intentional design.